cmd/htcondor-api/token_fetch.go

```diff
@@ -0,0 +1,433 @@
+// Package main provides token management functionality for the htcondor-api client
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+// TokenInfo stores OAuth2 token information for a trust domain
+type TokenInfo struct {
+	TrustDomain   string    `json:"trust_domain"`
+	ClientID      string    `json:"client_id"`
+	ClientSecret  string    `json:"client_secret"`
+	AccessToken   string    `json:"access_token"`
+	RefreshToken  string    `json:"refresh_token"`
+	TokenType     string    `json:"token_type"`
+	ExpiresAt     time.Time `json:"expires_at"`
+	Scopes        []string  `json:"scopes"`
+	IssuerURL     string    `json:"issuer_url"`
+}
+
+// TokenInfoStore manages multiple token infos keyed by trust domain
+type TokenInfoStore struct {
+	Tokens map[string]*TokenInfo `json:"tokens"`
+}
+
+// TokenFetchConfig holds configuration for fetching tokens
+type TokenFetchConfig struct {
+	IssuerURL   string
+	TrustDomain string
+	Scopes      []string
+	ClientID    string
+	ClientSecret string
+}
+
+// runTokenFetch implements the "token fetch" subcommand
+func runTokenFetch(args []string) error {
+	// Parse flags for token fetch
+	if len(args) < 1 {
+		return fmt.Errorf("usage: htcondor-api token fetch <issuer-url> [--trust-domain DOMAIN] [--scopes SCOPES]")
+	}
+
+	issuerURL := args[0]
+	trustDomain := ""
+	scopes := []string{"openid", "mcp:read", "mcp:write"}
```

Comment:
The scopes should include the various HTCondor ones (`READ`, `WRITE`, etc, becoming `condor:/READ`, `condor:/WRITE`, etc).